test/benchmark/simulator_benchmark.py

```diff
@@ -167,49 +163,12 @@ def add_expval(base, num_terms):
             circuit = circuit.bind_parameters(param_binds)
         
         simulator = self.simulators[runtime]
-        if measure == self.MEASUREMENT_SAMPLING:
-            if runtime in self.TRANSPLIERS:
-                runtime_circuit = eval(self.TRANSPLIERS[runtime])(circuit)
-                if (runtime, app, measure, measure_count, qubit) not in QOBJS:
-                    QOBJS[(runtime, app, measure, measure_count, qubit)] = assemble(runtime_circuit, simulator, shots=measure_count)
-                return QOBJS[(runtime, app, measure, measure_count, qubit)]
-            else:
-                runtime_circuit = circuit
-                if (simulator, app, measure, measure_count, qubit) not in QOBJS:
-                    QOBJS[(simulator, app, measure, measure_count, qubit)] = assemble(runtime_circuit, simulator, shots=measure_count)
-                return QOBJS[(simulator, app, measure, measure_count, qubit)]
-        elif measure == self.MEASUREMENT_EXPVAL:
-            if runtime in self.TRANSPLIERS:
-                runtime_circuit = eval(self.TRANSPLIERS[runtime])(circuit)
-                if (runtime, app, measure, measure_count, qubit) not in QOBJS:
-                    QOBJS[(runtime, app, measure, measure_count, qubit)] = assemble(runtime_circuit, simulator, shots=1)
-                return QOBJS[(runtime, app, measure, measure_count, qubit)]
-            else:
-                runtime_circuit = circuit
-                if (simulator, app, measure, measure_count, qubit) not in QOBJS:
-                    QOBJS[(simulator, app, measure, measure_count, qubit)] = assemble(runtime_circuit, simulator, shots=1)
-                return QOBJS[(simulator, app, measure, measure_count, qubit)]
+        shots = measure_count if measure == self.MEASUREMENT_SAMPLING else 1
+        
+        if (simulator, app, measure, measure_count, qubit) not in QOBJS:
+            QOBJS[(runtime, app, measure, measure_count, qubit)] = assemble(transpile(circuit, simulator), simulator, shots=shots)
```

Comment:
Do we want to be passing in qobjs manually to anything here as passing a qobj to `backend.run()` will emit a `DeprecationWarning`